onnxruntime/core/graph/signal_ops/signal_defs.cc

```diff
@@ -132,66 +123,93 @@ Generates a {name} window as described in the paper https://ieeexplore.ieee.org/
   };
 }
 
+ONNX_NAMESPACE::NodeProto WindowOpFunctionNode(const char* name) {
+  ONNX_NAMESPACE::NodeProto node;
+  node.set_op_type(std::string(name) + "Window");
+
+  auto* output_datatype = node.add_attribute();
+  output_datatype->set_name("output_datatype");
+  output_datatype->set_ref_attr_name("output_datatype");
+
+  auto* periodic = node.add_attribute();
+  periodic->set_name("periodic");
+  periodic->set_ref_attr_name("periodic");
+
+  node.add_input("size");
+  node.add_output("output");
+  return node;
+}
+
 void RegisterSignalSchemas() {
+  ONNX_NAMESPACE::NodeProto dft_function_node;
+  dft_function_node.set_op_type("DFT");
+
+  auto* dft_function_onesided = dft_function_node.add_attribute();
+  dft_function_onesided->set_name("onesided");
+  dft_function_onesided->set_ref_attr_name("onesided");
+
+  auto* dft_function_axis = dft_function_node.add_attribute();
+  dft_function_axis->set_name("axis");
+  dft_function_axis->set_ref_attr_name("axis");
+
+  auto* dft_function_inverse = dft_function_node.add_attribute();
+  dft_function_inverse->set_name("inverse");
+  dft_function_inverse->set_ref_attr_name("inverse");
+
+  dft_function_node.add_input("input");
+  dft_function_node.add_input("dft_length");
+  dft_function_node.add_output("output");
+
+  ONNX_NAMESPACE::OperatorSetIdProto onnx_op_set_17;
+  onnx_op_set_17.set_domain(kOnnxDomain);
+  onnx_op_set_17.set_version(17);
+
   MS_SIGNAL_OPERATOR_SCHEMA(DFT)
       .SetDomain(kMSExperimentalDomain)
       .SinceVersion(1)
       .SetDoc(R"DOC(DFT)DOC")
       .Attr("onesided",
-            "If True (default), only values for half of the fft size are returned because the real-to-complex Fourier transform satisfies the conjugate symmetry."
-            "The output tensor will return the first floor(n_fft/2) + 1 values from the DFT."
-            "Values can be 0 or 1.",
-            AttributeProto::AttributeType::AttributeProto_AttributeType_INT,
-            static_cast<int64_t>(0))
+            "If True (default), only values for half of the fft size are returned because the real-to-complex Fourier "
+            "transform satisfies the conjugate symmetry.The output tensor will return the first floor(n_fft/2) + 1 "
+            "values from the DFT. Values can be 0 or 1.",
+            AttributeProto::AttributeType::AttributeProto_AttributeType_INT, static_cast<int64_t>(0))
       .Attr("axis",
-            "The axis on which to perform the DFT. By default this value is set to 0, which corresponds to the first dimension after the batch index."
-            "This value must be less than signal_dimN, where signal_dimN is the number of dimensions in the signal.",
-            AttributeProto::AttributeType::AttributeProto_AttributeType_INT,
-            static_cast<int64_t>(0))
+            "The axis on which to perform the DFT. By default this value is set to 0, which corresponds to the first "
```

Comment:
Default is set to 1 by onnx.

Reply:
TODO: test ms experimental vs ONNX with both 1D and 2D inputs and default axis.

Reply:
And test `IDFT(DFT(x)) == x`